### PR TITLE
Access Log API Support for Virtual Gateways

### PR DESCRIPTION
**Pull request by rishijatia**
*Issue #, if available:* 37

*Description of changes:* 

Adding support for Logging per VirtualGateway.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by fawadkhaliq** [comment]
Can you please update the files here [1] as well. appmesh controller uses them to build preview sdk [1]

[1]https://github.com/aws/aws-app-mesh-roadmap/tree/master/appmesh-preview/sdk
[2] https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/preview/appmesh_models_override/setup.sh

**Comment by rishijatia** [comment]
Hi @fawadkhaliq, thank you for pointing that out. I have added the required files.

File: appmesh-preview/sdk/api.normal.json [hunk 1]
```diff
@@ -3282,12 +3287,15 @@
...
       "members": {
         "idle": {
-          "shape": "Duration"
+          "shape": "Duration",
+          "documentation": "<p>An object that represents an idle timeout. An idle timeout bounds the amount of time that a connection may be idle. The default value is none.</p>"
         },
         "perRequest": {
-          "shape": "Duration"
+          "shape": "Duration",
+          "documentation": "<p>An object that represents a per request timeout. The default value is 15 seconds. If you set a higher timeout, then make sure that the higher value is set for each App Mesh resource in a conversation. For example, if a virtual node backend uses a virtual router provider to route to another virtual node, then the timeout should be greater than 15 seconds for the source and destination virtual node and the route.</p>"
```
Comment (alexbrjo) [inline appmesh-preview/sdk/api.normal.json:3295]: The "for the source and destination" could use some clarification. Shouldn't it be the other way around? This is what I get from the description: 

`[VN w/ backend timeout=20] --> [VS] --> [VR + Route] --> [VN timeout=25]`

But logically it seems like the VN using the other service as a backend should have a larger timeout. Like this:

`[VN w/ backend timeout=25] --> [VS] --> [VR + Route] --> [VN timeout=20]`

Comment (rishijatia) [reply appmesh-preview/sdk/api.normal.json:3295]: I think this is auto-generated from our doc-model, can take this discussion offline.

**Pull request merged**